graph.cpp

```diff
@@ -65,12 +67,12 @@ void Graph::removeNode(const std::string& name) {
         node->disconnectAllStreams();
         
         // Make sure to unlock before "node" is destroyed.
-        lock.unlock();
+        lock = nullptr;
```

Comment:
This is probably the strangest part of the PR: to be able to unlock the lock_guard we need to destroy it, so I'm placing it in the heap.

Reply:
interesting solution... but maybe simply calling mutex_.lock() and mutex_.unlock() would be better here...

Reply:
I'd argue that a RAII-based way is safer: if a function throws then the mutex will remain locked (not that it happens now, but it can easily happen later as we extend the library)

Reply:
hum this theoretical additional safety comes at a performance price. But since this method does not need to be fast, let's keep your safe solution.